miniscope_io/stream_daq.py

```diff
@@ -51,11 +52,14 @@ def exact_iter(f: Callable, sentinel: Any) -> Generator[Any, None, None]:
     because truth value of numpy arrays is ambiguous.
     """
     while True:
-        val = f()
-        if val is sentinel:
-            break
-        else:
-            yield val
+        try:
+            val = f()
+            if val is sentinel:
+                break
+            else:
+                yield val
+        except queue.Empty:
+            pass
```

Comment:
Added handling of `queue.Empty` here. This kind of assumes the data is a queue and feels less generic function, but also thought making the `try` block double-layered in all functions make stuff complex.

Reply:
Ok I am now a *little* confused at what we are going for in this PR - 

in the existing implementation, we are iterating over a Queue.get call. Without a timeout, that should block indefinitely until something is in the queue. queue.Empty exceptions should only happen if we add a timeout in the get() call, which I dont think we are doing here. All the downstream queues are fine to wait indefinitely for input, because the first fpga_recv method always terminates by putting a None in the queue which terminates the iteration.

Am I right that the purpose of the PR is to keep capturing through errors? I think that the only change necessary for that is the change to the StreamReadError handling. The downstream methods just wouldnt notice and would keep processing whatever gets put in the queues.

I thought part of the goal was to make it so the queues arent strict FIFO queues that might fall behind realtime, and instead just work off the most recent data, hence the timeouts on the `put()` calls. But if thats not the goal then im not sure what those are for?

So it seems like there are a few things going on here and im not sure which of them are desired behavior and which of them might be leftovers from prior attempts.

Reply:
Oh I was confused about this part too and you're totally right. I guess fixing it around blocking/no-timeout get and blocking/timeout put to keep it continously work kind of interactively will do. I'll remove this exception handling and merge when I confirm it works.

Reply:
Ok lmk, I can help with tidying up if there were a few different tries here!